### ISSUE TITLE
Tagged template has been broken since v20200315

### ISSUE THREAD

**Issue by teppeis** [google/closure-compiler#3589]
Input:
```js
function tag(args) {
  console.log(args);
}
tag`aaa`;
```

Output (v20200426, SIMPLE):
```js
var $jscomp$templatelit$2052066224$0 = $jscomp.createTemplateTagFirstArg(["aaa"]), $jscomp = $jscomp || {};
$jscomp.scope = {};
$jscomp.createTemplateTagFirstArg = function(a) {
  return a.raw = a;
};
$jscomp.createTemplateTagFirstArgWithRaw = function(a, b) {
  a.raw = b;
  return a;
};
function tag(a) {
  console.log(a);
}
tag($jscomp$templatelit$2052066224$0);
```

Runtime error (in Node v12.16.3):
```
var $jscomp$templatelit$2052066224$0 = $jscomp.createTemplateTagFirstArg(["aaa"]), $jscomp = $jscomp || {};
                                               ^

TypeError: Cannot read property 'createTemplateTagFirstArg' of undefined
    at Object.<anonymous> (dist/bundle.js:1:48)
    at Module._compile (internal/modules/cjs/loader.js:1133:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32)
    at Function.Module._load (internal/modules/cjs/loader.js:877:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47
```
---

Correct output (v20200224):
```js
var $jscomp$templatelit$0 = ["aaa"];
$jscomp$templatelit$0.raw = $jscomp$templatelit$0.slice();
function tag(a) {
  console.log(a);
}
tag($jscomp$templatelit$0);
```

**Comment by brad4d** [comment]
I think I know what's going on here.

1. The definitions of `$jscomp` and its methods are automatically injected at the start of the first input script / module.
2. When we transpile the template tag, we must declare the template tag arguments at the script / module level, so we put the declaration at the top of the current script / module.
3. Unfortunately, the tag is in the first script / module, so we end up putting the declaration before the definitions it depends on.

I believe the best fix is instead to traverse up the AST from the tagged template to find the module - level statement containing it and insert the declaration at that point instead.

Rishi, would you mind making that change?

**Comment by rishipal** [comment]
Fixed with cedbad5941e5adaeb4e22c8009be4478eb0bd82e
